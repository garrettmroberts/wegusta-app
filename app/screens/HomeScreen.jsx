import React from 'react';
import { View, FlatList, ScrollView, Text, StyleSheet } from 'react-native';
import { useStoreContext } from '../utils/Context';
import { colors } from '../config';
import { Feather, Ionicons } from '@expo/vector-icons';

import { Button, SocialButton, IconButton, DecisionButton, OverlayButton, SearchInput } from '../components';

const handlePress = () => { console.log('pressed'); };
const HomeScreen = () => {
  const [state, dispatch] = useStoreContext();

  const addPersonIcon = <Ionicons name="person-add" size={24} color="white" />;
  const addPersonIconSmall = <Ionicons name="person-add" size={11} color="white" />;

  return (
    <View style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollContainer}>
        <View>
          <Text style={styles.header}>Colors</Text>
          <View style={styles.section}>
            <View style={[styles.colorSquare, { backgroundColor: colors.primary }]} >
              <Text style={{ color: colors.white }}>primary</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.greyDark }]} >
              <Text>greyDark</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.grey }]} >
              <Text>grey</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.greyLight }]} >
              <Text>greyLight</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.white }]} >
              <Text>white</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.success }]} >
              <Text>success</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.warning }]} >
              <Text>warning</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.error }]} >
              <Text>error</Text>
            </View>
            <View style={[styles.colorSquare, { backgroundColor: colors.facebookBlue }]} >
              <Text>facebookBlue</Text>
            </View>
          </View>
        </View>
        <Button type="primary" size="fullWidth" iconPlacement="left" text="Codename LLC" handlePress={handlePress} icon="person-add" />
        <Button type="secondary" size="large" iconPlacement="right" text="Codename LLC" handlePress={handlePress} icon="person-add" />
        <Button type="disabled" size="medium" iconPlacement="both" text="Codename LLC" handlePress={handlePress} icon="person-add" />
        <Button type="destructive" size="small" text="Codename LLC" handlePress={handlePress} icon="person-add" />

        <SocialButton icon="facebook" handlePress={handlePress} />
        <DecisionButton decision="like" />
        <DecisionButton decision="dislike" />
        <OverlayButton dark={true} overlay="more" />
        <OverlayButton dark={false} overlay="search" />
        <OverlayButton dark={true} overlay="close" />
        <SearchInput placeholder='Search' />
        <SearchInput placeholder='Search' disabled={true} />
        <IconButton size="large" color="primary" icon="person-add" handlePress={handlePress} />
        <IconButton size="medium" color="primary" icon="person-add" handlePress={handlePress} />
        <IconButton size="small" color="error" icon="person-add" handlePress={handlePress} />
        <IconButton size="xsmall" color="error" icon="person-add" handlePress={handlePress} />
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    fontSize: 20,
    color: colors.white,
    fontWeight: '700'
  },
  // eslint-disable-next-line react-native/no-color-literals
  scrollContainer: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  // eslint-disable-next-line react-native/no-color-literals
  container: {
    flex: 1,
    backgroundColor: '#6546F9'
  },
  section: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    marginBottom: 15

  },
  colorSquare: {
    height: 70,
    width: 70,
    margin: 2
  }
});

export default HomeScreen;
