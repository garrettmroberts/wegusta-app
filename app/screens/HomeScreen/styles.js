import { StyleSheet } from 'react-native';
import { colors, constants, fonts } from '../../config';

const styles = StyleSheet.create({
  navHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    marginRight: 24
  },
  navIconGap: {
    width: 24
  }
});

export default styles;
