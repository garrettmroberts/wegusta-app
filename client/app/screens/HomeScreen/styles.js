import { StyleSheet } from 'react-native'
import { constants, fonts } from '../../config'

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    flex: 1,
    justifyContent: 'center',
  },
  wrapper: {
    marginLeft: 24,
    alignItems: 'flex-start',
    maxWidth: constants.screenWidth - constants.screenWidth * 0.46,
    position: 'relative',
    bottom: 60,
  },
  header: {
    ...fonts.displayBold,
    ...fonts.textDark,
  },
  secondaryHeader: {
    ...fonts.displayBold,
    ...fonts.textGreyDark,
  },
  subtext: {
    ...fonts.h2Reg,
    ...fonts.textDark,
    marginBottom: 16,
  },
})

export default styles
