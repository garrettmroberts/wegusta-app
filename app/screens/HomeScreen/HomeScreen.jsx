import React from 'react';
import { View, ImageBackground, Text } from 'react-native';

import Button from '../../components/Button/Button';
import styles from './styles';

const HomeScreen = () => {
  return (
    <View style={styles.container}>
      <ImageBackground source={require('../../assets/homescreenBackground.png')} resizeMode="cover" style={styles.image}>
        <View style={styles.wrapper}>
          <Text style={styles.header}>No more nose goes...</Text>
          <Text style={styles.subtext}>Find the perfect restaurant for everyone.</Text>
          <Button type='primary' size='large' text='Make Plans' iconPlacement="right" icon="arrow-forward" />
        </View>
      </ImageBackground>
    </View>
  );
};

export default HomeScreen;
