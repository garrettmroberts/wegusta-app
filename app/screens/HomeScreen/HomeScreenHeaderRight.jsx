import React from 'react';
import { View } from 'react-native';

import Avatar from '../../components/Avatar/Avatar';
import { colors } from '../../config';
import buildIcon from '../../utils/buildIcon';

import styles from './styles';
import { useNavigation } from '@react-navigation/native';

const HomeScreenHeaderRight = () => {
  const navigation = useNavigation();

  return (
    <View style={styles.navHeader} >
      { buildIcon({
        name: 'notifications-outline',
        color: colors.black,
        size: 32,
        onPress: () => {navigation.navigate('ComponentsDemo');}
      })}
      <View style={styles.navIconGap} />
      <Avatar
        size='navSized'
      />
    </View>
  );
};

export default HomeScreenHeaderRight;
