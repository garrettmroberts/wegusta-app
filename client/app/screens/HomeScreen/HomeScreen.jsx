import React from 'react'
import { View, ImageBackground, Text } from 'react-native'
import { PropTypes } from 'prop-types'

import Button from '../../components/Button/Button'
import styles from './styles'

const HomeScreen = ({ navigation }) => {
    const navigateButton = () => {
        navigation.navigate('Invite')
    }

    return (
        <View style={styles.container}>
            <ImageBackground
                source={require('../../assets/homescreenBackground.png')}
                resizeMode="cover"
                style={styles.image}
            >
                <View style={styles.wrapper}>
                    <Text style={styles.header}>We Like,</Text>
                    <Text style={styles.secondaryHeader}>together</Text>
                    <Text style={styles.subtext}>
                        Find the perfect restaurant for everyone.
                    </Text>
                    <Button
                        type="primary"
                        size="large"
                        text="Make Plans"
                        iconPlacement="right"
                        icon="arrow-forward"
                        handlePress={navigateButton}
                    />
                </View>
            </ImageBackground>
        </View>
    )
}

HomeScreen.propTypes = {
    navigation: PropTypes.object,
}

export default HomeScreen
