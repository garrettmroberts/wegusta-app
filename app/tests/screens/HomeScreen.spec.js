import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../../screens/HomeScreen.jsx';

test('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
