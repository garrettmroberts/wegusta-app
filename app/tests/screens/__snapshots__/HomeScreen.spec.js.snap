// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEF0F4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Foodie.io
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "opacity": 1,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#6546F9",
          "borderRadius": 12,
          "height": 48,
          "justifyContent": "center",
          "width": 702,
        },
      ]
    }
    text="Ballyhoo"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6546F9",
          "borderRadius": 12,
          "height": 48,
          "justifyContent": "center",
          "width": 702,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Avenir",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "900",
            "letterSpacing": 0,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Ballyhoo
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "opacity": 1,
        },
        Object {
          "alignItems": "center",
          "borderRadius": 40,
          "elevation": 4,
          "height": 80,
          "justifyContent": "center",
          "margin": 16,
          "shadowColor": "#1C262D",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
          "width": 80,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 40,
          "elevation": 4,
          "height": 80,
          "justifyContent": "center",
          "margin": 16,
          "shadowColor": "#1C262D",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2.62,
          "width": 80,
        }
      }
    />
  </View>
</View>
`;
